### PR TITLE
adding annotation for kube-aggregator automanaged in labels-md

### PR DESCRIPTION
**Pull request by Ritikaa96**
Code from the Kubernetes project sets (or previously did set) a label, kube-aggregator.kubernetes.io/automanaged

This looks intended, but https://kubernetes.io/docs/reference/labels-annotations-taints/ doesn't document that label.
This PR document & register the label .

Page to Update:
https://kubernetes.io/docs/reference/labels-annotations-taints/

#38126

### CODE REVIEW

**Comment by sftim** [comment]
/sig api-machinery

I think

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -155,6 +155,20 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
...
+
+Used on: APIService
+
+It is a label attached to the APIService that identifies how the APIService wants to be synced. `manageOnStart` is a value for the AutoRegisterManaged label that indicates the APIService wants to be synced one time when the controller starts.
+
+`manageOnStart = "onstart"`
+
+`manageContinuously` is a value for the AutoRegisterManaged label that indicates the APIService wants to be synced continuously.
+
+`manageContinuously = "true"`
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:170]: This looks like a direct quote from the code. I'd prefer to rephrase it so it makes sense to someone who doesn't understand even a symbol of Go.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -155,6 +155,20 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
 
 This label has been deprecated. Please use `kubernetes.io/os` instead.
 
+### kube-aggregator.kubernetes.io/automanaged {kube-aggregator-kubernetesio-automanaged}
+
+Example: `kube-aggregator.kubernetes.io/automanaged: "manageContinuously"`
```
[the remark this answers is not in the document]
Comment (Ritikaa96) [reply content/en/docs/reference/labels-annotations-taints/_index.md:160]: Noted.

**Comment by Ritikaa96** [comment]
@sftim Made the changes as per the comments, PTAL!

**Comment by Ritikaa96** [comment]
Hi @sftim @bradtopol @natalisucks I hope all the nits have been resolved . please take a look at the PR

**Comment by natalisucks** [comment]
@kubernetes/sig-api-machinery-misc Howdy folks, would this change be in your realm for review? We're looking for a tech review on the above addition. Thanks in advance!

**Comment by Ritikaa96** [comment]
Thanks, @natalisucks 
@kubernetes/sig-api-machinery-misc waiting for further instructions or suggestions.

**Comment by enj** [comment]
For context:

https://github.com/kubernetes/kubernetes/pull/51921

https://github.com/kubernetes/kubernetes/issues/51912

Not sure what level of detail is necessary here.

**Comment by sftim** [comment]
This level of detail looks good. If we wanted to explain more, that'd happen on a different page.

**Comment by sftim** [comment]
What we want is a tech review to confirm that this change is not misleading to readers.

**Comment by lavalamp** [comment]
For tech review:

/assign @deads2k @liggitt

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -155,6 +155,18 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
 
 This label has been deprecated. Please use `kubernetes.io/os` instead.
 
+### kube-aggregator.kubernetes.io/automanaged {kube-aggregator-kubernetesio-automanaged}
+
+Example: `kube-aggregator.kubernetes.io/automanaged: "onstart"`
+
+Used on: APIService
+
+This label is attached to the APIService to identify how the APIService wants to be synced.
```
[the remark this answers is not in the document]
Comment (liggitt) [reply content/en/docs/reference/labels-annotations-taints/_index.md:164]: Maybe add a sentence explaining what `kube-apiserver` does with automanaged APIService objects:

> Automanaged APIService objects are deleted by `kube-apiserver` when it has no built-in or custom resource API corresponding to the API group/version of the APIService.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -155,6 +155,18 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
...
+### kube-aggregator.kubernetes.io/automanaged {kube-aggregator-kubernetesio-automanaged}
+
+Example: `kube-aggregator.kubernetes.io/automanaged: "onstart"`
+
+Used on: APIService
+
+This label is attached to the APIService to identify how the APIService wants to be synced. 
+
+If the label is set to  `onstart`, it indicates that the APIService wants to be synced one time i.e. when the controller starts.
+To indicate that the APIService wants to be synced continuously the label is set to `true`.
```
Comment (liggitt) [inline content/en/docs/reference/labels-annotations-taints/_index.md:167]: These *really* aren't intended for users to set/modify, so I'm not sure how much we want to document here, but if we do document this, we should avoid anthropomorphizing the controller... it doesn't like it ;-)

Either drop this bit or maybe rephrase like:

> Possible values are:
> * `onstart` indicates the APIService should be reconciled once at start
> * `true` indicates the APIService should be reconciled continuously

Comment (Ritikaa96) [reply content/en/docs/reference/labels-annotations-taints/_index.md:167]: Got it @liggitt , thanks for the suggestions.

**Comment by Ritikaa96** [comment]
Hi @liggitt , made the changes as per suggestions, hope this is workable now. PTAL

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 5]
```diff
@@ -155,6 +155,23 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
 
 This label has been deprecated. Please use `kubernetes.io/os` instead.
 
+### kube-aggregator.kubernetes.io/automanaged {kube-aggregator-kubernetesio-automanaged}
+
+Example: `kube-aggregator.kubernetes.io/automanaged: "onstart"`
+
+Used on: APIService
+
+This label is attached to the APIService objects created by `kube-apiserver` to identify how they should be managed by `kube-apiserver`. This label should not be added to other APIService objects or modified on managed APIService objects.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:164]:
```suggestion
The kube-apiserver sets this label on any APIService objects that the API server has created
automatically. The label marks how the control plane should manage that APIService.
You should not add, modify, or remove this label yourself.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 6]
```diff
@@ -155,6 +155,23 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
...
+
+{{< note >}}
+Automanaged APIService objects are deleted by kube-apiserver when it has no built-in or custom resource API corresponding to the API group/version of the APIService.
+{{< /note >}}
+
+It has two possible values:
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:170]: (nit)
```suggestion
There are two possible values:
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 7]
```diff
@@ -155,6 +155,23 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
...
+* `onstart` indicates the APIService should be reconciled once at start.
+* `true` indicates the APIService should be reconciled continuously.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:172]: I'd use a definition list; for example:
```suggestion

`onstart`
: the APIService should be reconciled when an API server starts up, but not otherwise.

`true`
: the API server should reconcile this APIService continuously.
```

**Review by sftim — COMMENTED** [review]
Nearly LGTM. The fragment identifier fix is essential; the other changes are just nice to have.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 8]
```diff
@@ -155,6 +155,23 @@ This label has been deprecated. Please use `kubernetes.io/arch` instead.
 
 This label has been deprecated. Please use `kubernetes.io/os` instead.
 
+### kube-aggregator.kubernetes.io/automanaged {kube-aggregator-kubernetesio-automanaged}
```
[the remark this answers is not in the document]
Comment (Ritikaa96) [reply content/en/docs/reference/labels-annotations-taints/_index.md:158]: right. i might have missed this one . thanks @sftim

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 5 above)
Comment (Ritikaa96) [reply content/en/docs/reference/labels-annotations-taints/_index.md:164]: noted

**Comment by Ritikaa96** [comment]
@sftim @liggitt Hi! Updated the content with suggested changes . PTAL!

**Comment by natalisucks** [comment]
Thanks @Ritikaa96 for applying the suggestions. From a docs perspective, this looks good to me. Once we have the final tech review sign off I'd be happy to approve.

**Comment by tengqm** [comment]
We have approvals from docs and tech sides.
/lgtm
/approve

**Comment by Ritikaa96** [comment]
Thanks for the approval @tengqm and @natalisucks
